src/test/java/bio/terra/service/snapshotbuilder/SnapshotBuilderServiceTest.java

```diff
@@ -37,4 +40,38 @@ void createSnapshotRequest() {
             datasetId, SnapshotBuilderTestData.createSnapshotAccessRequest(), email),
         response);
   }
+
+  @Test
+  void enumerateSnapshotRequestsByDatasetId() {
+    UUID datasetId = UUID.randomUUID();
+    SnapshotAccessRequestResponse responseItem = new SnapshotAccessRequestResponse();
+    List<SnapshotAccessRequestResponse> response = List.of(responseItem, responseItem);
+    when(snapshotRequestDao.enumerateByDatasetId(datasetId)).thenReturn(response);
+    Assertions.assertEquals(
+        snapshotBuilderService.enumerateByDatasetId(datasetId),
+        snapshotBuilderService.convertToEnumerateModel(response));
+  }
+
+  @Test
+  void convertToEnumerateModel() {
+    SnapshotAccessRequestResponse inputItem =
+        SnapshotBuilderTestData.createSnapshotAccessRequestResponse();
+    List<SnapshotAccessRequestResponse> response = List.of(inputItem);
+
+    EnumerateSnapshotAccessRequestItem expectedItem = new EnumerateSnapshotAccessRequestItem();
+    expectedItem.id(inputItem.getId());
+    expectedItem.status(inputItem.getStatus());
+    expectedItem.createdDate(inputItem.getCreatedDate());
+    expectedItem.name(inputItem.getSnapshotName());
+    expectedItem.researchPurpose(inputItem.getSnapshotResearchPurpose());
+    EnumerateSnapshotAccessRequest expected = new EnumerateSnapshotAccessRequest();
+    expected.addItemsItem(expectedItem);
+
+    EnumerateSnapshotAccessRequest converted =
+        snapshotBuilderService.convertToEnumerateModel(response);
+    EnumerateSnapshotAccessRequestItem convertedItem = converted.getItems().get(0);
+
+    Assertions.assertEquals(converted.getItems().size(), expected.getItems().size());
+    Assertions.assertEquals(convertedItem, expectedItem);
```

Comment:
Here is another way to write these assertions, which I think gets closer to the spirit of what you're wanting to assert and avoids a possible NPE:

```
var converted = snapshotBuilderService.convertToEnumerateModel(response).getItems();
assertThat(items, hasSize(1));
assertThat(items.get(0), equalTo(expectedItem));
```

Or even more concisely (and my preference):
```
assertThat(snapshotBuilderService.convertToEnumerateModel(response).getItems(), contains(expectedItem));
```

Relevant Hamcrest docs:
- https://hamcrest.org/JavaHamcrest/javadoc/1.3/org/hamcrest/Matchers.html#hasSize(int)
- https://hamcrest.org/JavaHamcrest/javadoc/1.3/org/hamcrest/Matchers.html#contains(E...)

Reply:
I agree, although I usually use `containsInAnyOrder()` instead of `contains()`, if your test doesn't need to assert on the order of elements in the list. If the order _is_ fixed and you want to assert on it, you can also compare the entire model:
```
    assertThat(converted, is(expected));
```